### PR TITLE
Change Android login flow (MS case 19470)

### PR DESCRIPTION
**Pull request by gcalero**
https://highfidelity.manuscript.com/f/cases/19470

This PR implements the new login flow, according to the provided wireframe
https://www.figma.com/file/q9L93vRFZU4vW6alHLH5gy/Mobile-Wireframe-Beta?node-id=0%3A1

OAuth2 is implemented using a client based on these 3 parameters that needs to be passed as gradle parameters for android build: **OAUTH_CLIENT_ID**, **OAUTH_CLIENT_SECRET**, **OAUTH_REDIRECT_URI**

Test plan
1) The app encourage the user to login when the app starts
2) User is able to create an account in HighFidelity
3) User is able to login with username/email + password
4) User is able to login through steam account
5) User may skip the login and go back later
6) Test navigation across all screens (including back navigation)
7) Test entering to domains that require authentication
8) Test validations and connectivity errors during login & signup

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14416/HighFidelity-Beta-PR14282-11d82e0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14416/HighFidelity-Beta-PR14282-11d82e0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14416/HighFidelity-Beta-Interface-PR14282-11d82e0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14416/HighFidelity-Beta-PR14282-11d82e0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14416/HighFidelity-Beta-Interface-PR14282-11d82e0.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14425/HighFidelity-Beta-PR14282-3d677d8.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14425/HighFidelity-Beta-PR14282-3d677d8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14425/HighFidelity-Beta-Interface-PR14282-3d677d8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14425/HighFidelity-Beta-PR14282-3d677d8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14425/HighFidelity-Beta-Interface-PR14282-3d677d8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14428/HighFidelity-Beta-PR14282-a1221eb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14428/HighFidelity-Beta-Interface-PR14282-a1221eb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14428/HighFidelity-Beta-PR14282-a1221eb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14428/HighFidelity-Beta-Interface-PR14282-a1221eb.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14659/HighFidelity-Beta-PR14282-6a3ccbb.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14659/HighFidelity-Beta-PR14282-6a3ccbb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14659/HighFidelity-Beta-Interface-PR14282-6a3ccbb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14659/HighFidelity-Beta-PR14282-6a3ccbb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14659/HighFidelity-Beta-Interface-PR14282-6a3ccbb.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14718/HighFidelity-Beta-PR14282-9ed4374.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14718/HighFidelity-Beta-PR14282-9ed4374.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14718/HighFidelity-Beta-Interface-PR14282-9ed4374.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14718/HighFidelity-Beta-PR14282-9ed4374.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14718/HighFidelity-Beta-Interface-PR14282-9ed4374.exe

**Comment by gcalero** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14849/HighFidelity-Beta-PR14282-9ed4374.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14849/HighFidelity-Beta-Interface-PR14282-9ed4374.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14849/HighFidelity-Beta-PR14282-9ed4374.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14849/HighFidelity-Beta-Interface-PR14282-9ed4374.exe

**Comment by gcalero** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14857/HighFidelity-Beta-PR14282-9ed4374.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14857/HighFidelity-Beta-PR14282-9ed4374.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14857/HighFidelity-Beta-Interface-PR14282-9ed4374.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14878/HighFidelity-Beta-PR14282-ccd0d82.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14878/HighFidelity-Beta-PR14282-ccd0d82.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14878/HighFidelity-Beta-Interface-PR14282-ccd0d82.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14878/HighFidelity-Beta-PR14282-ccd0d82.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14878/HighFidelity-Beta-Interface-PR14282-ccd0d82.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14918/HighFidelity-Beta-PR14282-1d059d9.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14918/HighFidelity-Beta-PR14282-1d059d9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14918/HighFidelity-Beta-Interface-PR14282-1d059d9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14924/HighFidelity-Beta-PR14282-79a71d2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14924/HighFidelity-Beta-PR14282-79a71d2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14924/HighFidelity-Beta-Interface-PR14282-79a71d2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14924/HighFidelity-Beta-PR14282-79a71d2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14924/HighFidelity-Beta-Interface-PR14282-79a71d2.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14950/HighFidelity-Beta-PR14282-79a71d2.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14952/HighFidelity-Beta-PR14282-c1174ad.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14952/HighFidelity-Beta-PR14282-c1174ad.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14952/HighFidelity-Beta-Interface-PR14282-c1174ad.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14952/HighFidelity-Beta-PR14282-c1174ad.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14952/HighFidelity-Beta-Interface-PR14282-c1174ad.exe

File: android/app/src/main/java/io/highfidelity/hifiinterface/fragment/LoginFragment.java [hunk 1]
```diff
@@ -104,13 +146,60 @@ public void onResume() {
     @Override
     public void onStop() {
         super.onStop();
-        cancelActivityIndicator();
+        if (!mUseOauth) {
+            showLoginForm();
+        }
```
Comment (SamGondelman) [inline android/app/src/main/java/io/highfidelity/hifiinterface/fragment/LoginFragment.java]: Why do we do this?

File: android/app/src/main/java/io/highfidelity/hifiinterface/fragment/LoginFragment.java [hunk 2]
```diff
@@ -2,53 +2,88 @@
...
+public class LoginFragment extends Fragment
+                            implements  OnBackPressedListener {
+
+    private static final String ARG_USE_OAUTH = "use_oauth";
+    private static final String TAG = "Interface";
+
+    private final String OAUTH_CLIENT_ID = BuildConfig.OAUTH_CLIENT_ID;
+    private final String OAUTH_REDIRECT_URI = BuildConfig.OAUTH_REDIRECT_URI;
+    private final String OAUTH_AUTHORIZE_BASE_URL = "https://highfidelity.com/oauth/authorize";
+    private static final int OAUTH_AUTORIZE_REQUEST = 1;
```
Comment (SamGondelman) [inline android/app/src/main/java/io/highfidelity/hifiinterface/fragment/LoginFragment.java]: `AUTHORIZE`?

File: android/app/src/main/java/io/highfidelity/hifiinterface/fragment/SignupFragment.java [hunk 3]
```diff
@@ -101,15 +114,23 @@ public void onResume() {
     @Override
     public void onStop() {
         super.onStop();
-        cancelActivityIndicator();
+        showSignupForm();
```
Comment (SamGondelman) [inline android/app/src/main/java/io/highfidelity/hifiinterface/fragment/SignupFragment.java]: ?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14995/HighFidelity-Beta-PR14282-729a0b6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14995/HighFidelity-Beta-PR14282-729a0b6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14995/HighFidelity-Beta-Interface-PR14282-729a0b6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14995/HighFidelity-Beta-PR14282-729a0b6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14995/HighFidelity-Beta-Interface-PR14282-729a0b6.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14996/HighFidelity-Beta-PR14282-d9f68af.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14997/HighFidelity-Beta-PR14282-7687eb9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14997/HighFidelity-Beta-PR14282-7687eb9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14997/HighFidelity-Beta-Interface-PR14282-7687eb9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14997/HighFidelity-Beta-PR14282-7687eb9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14997/HighFidelity-Beta-Interface-PR14282-7687eb9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15012/HighFidelity-Beta-PR14282-4f4239f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15012/HighFidelity-Beta-Interface-PR14282-4f4239f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15012/HighFidelity-Beta-PR14282-4f4239f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15012/HighFidelity-Beta-Interface-PR14282-4f4239f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15036/HighFidelity-Beta-PR14282-90295a0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15036/HighFidelity-Beta-PR14282-90295a0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15036/HighFidelity-Beta-Interface-PR14282-90295a0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15036/HighFidelity-Beta-PR14282-90295a0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15036/HighFidelity-Beta-Interface-PR14282-90295a0.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15049/HighFidelity-Beta-PR14282-0aaf92c.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15049/HighFidelity-Beta-PR14282-0aaf92c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15049/HighFidelity-Beta-Interface-PR14282-0aaf92c.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15066/HighFidelity-Beta-PR14282-e9fe0c8.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15066/HighFidelity-Beta-PR14282-e9fe0c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15066/HighFidelity-Beta-Interface-PR14282-e9fe0c8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15066/HighFidelity-Beta-PR14282-e9fe0c8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15066/HighFidelity-Beta-Interface-PR14282-e9fe0c8.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
The Steam Log In flow takes app to log in via High Fidelity website. Flow shown below with logs.
flow
[SteamFlow.txt](https://github.com/highfidelity/hifi/files/2577044/SteamFlow.txt)

**Pull request merged**